subsys/portability/cmsis_rtos_v2/thread.c

```diff
@@ -38,6 +38,10 @@ static K_THREAD_STACK_ARRAY_DEFINE(cmsis_rtos_thread_stack_pool,
 				   CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE);
 #endif
 
+struct stack_usage_info {
+	bool is_in_use;
+} stack_status[CONFIG_CMSIS_V2_THREAD_DYNAMIC_MAX_COUNT];
```

Comment:
This declaration is closely related to `cmsis_rtos_thread_stack_pool`. So, I suggest:
  - place it just after `cmsis_rtos_thread_stack_pool` in the `if CONFIG_CMSIS_V2_THREAD_DYNAMIC_MAX_COUNT != 0` block (even if this condition looks useless, I agree).
  - I wonder if declaring a struct with only one member is not just increasing complexity for free.
  - `stack_usage_info` vs `stack_status` vs `is_in_use`. Having several names for the same purpose make code more complex.
  - change the name to outline the link with `cmsis_rtos_thread_stack_pool`. eg. cmsis_rtos_thread_stack_in_use.

Therefore, I would use `bool cmsis_rtos_thread_stack_in_use[CONFIG_CMSIS_V2_THREAD_DYNAMIC_MAX_COUNT]`.